extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/RestDataProcessor.java

```diff
@@ -57,4 +63,36 @@ JsonbSerializerBuildItem registerJsonbSerializers() {
     RuntimeInitializedClassBuildItem el() {
         return new RuntimeInitializedClassBuildItem(EL.class.getCanonicalName());
     }
+
+    private void verifyHalResources(IndexView index, List<RestDataResourceBuildItem> resourceBuildItems) {
+        if (!hasJsonProcessor() && hasHalResource(index, resourceBuildItems)) {
+            logger.warn("Cannot generate HAL endpoints without a serializer. "
+                    + "Make sure you have either Jsonb or Jackson RESTEasy extension enabled.");
+            throw new RestDataBuildException("Cannot generate HAL endpoints without a serializer");
+        }
+    }
+
+    private boolean hasHalResource(IndexView index, List<RestDataResourceBuildItem> resourceBuildItems) {
+        ResourcePropertiesAccessor propertiesAccessor = new ResourcePropertiesAccessor(index);
+        for (RestDataResourceBuildItem resource : resourceBuildItems) {
+            if (propertiesAccessor.isHal(resource.getResourceInfo().getClassInfo())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean hasJsonProcessor() {
```

Comment:
You should rely on the `Capabilities` here. The ones existing don't really cover your use case (I checked, even if their names sound appealing, they are not covering your needs) so you will need to add `RESTEASY_JACKSON` and `RESTEASY_JSONB` and produce them from the respective processors together with the others.